### DISCUSSION TITLE
Particle spawning probality

### DISCUSSION THREAD

**Discussion by timotheeficat** [4ian/GDevelop#3532] [Enhancement ideas (and their technical discussions)]
I wanted to create stars in the background of my game so I created a particles emitter which has a spray cone angle of 360 degrees and a screen width sized radius.
But, when the particles spawn, there is more particles in the center of the screen.
I made a screenshot of it:
![image](https://user-images.githubusercontent.com/1670670/149674151-b3abc68a-87ea-40c9-a5a5-e1cde08515af.png)

Is there a way to prevent this?
Thanks!

**Comment by 4ian** [comment]
Can you post a screenshot of the settings of your particle emitter? :)

**Comment by timotheeficat** [comment]
![image](https://user-images.githubusercontent.com/1670670/149674157-f44f3042-cf62-46f5-832d-c49d9675bc53.png)
There are multiple emitters like this to create a parallax effect with multiple cameras.

**Comment by Lizard-13** [comment]
Well, I've made a particles engine myself before, and I may know what's happening here. Probably the circle spawner chooses a new particle location as anybody would do:
```
angle = Random(min_angle, max_angle)
distance = Random(min_radius, max_radius)
x = distance * cos(angle)
y = distance * sin(angle)
```
And although it seems right, the probability for a particule to be spawned in the most exterior ring (max_radius) is the same than the probability to be spawned in the most inner ring (min_radius), so you'll have approximately the same amount of particles in the outer ring and in the inner ring, the last has less radius hence you'll get a greater density of particles.

To solve it we must reduce the probability for smaller radiuses and increase the probability for the outer ones. There are more than one way to solve it, and one that seems much faster than the one I used in my little engine :open_mouth: 
Lets see if the library hasn't been updated first :smiley:

**Comment by AlexandreSi** [comment]
Hi there!
Working on fixing a few things around particle emitter, i can see that [the pixi renderer we use](https://pixijs.io/particle-emitter/docs/classes/Emitter.html#particlesPerWave) does not support the kind of method @Lizard-13 suggests (see spawnChance).
I don't think that the particle emitter is the solution to your issue @timotheeficat. I don't think there's currently a way of achieving your goal but we could integrate [Pixi's Particle container](https://pixijs.download/dev/docs/PIXI.ParticleContainer.html) in the future.
